### DISCUSSION TITLE
What's the difference between Entity and Model?

### DISCUSSION THREAD

**Discussion by alper** [SeaQL/sea-orm#999] [General]
I still don't get it.

And why do I get this unused stuff in the generated file?

```
#[derive(Copy, Clone, Debug, EnumIter)]
pub enum Relation {}

impl RelationTrait for Relation {
    fn def(&self) -> RelationDef {
        panic!("No RelationDef")
    }
}

impl ActiveModelBehavior for ActiveModel {}
```

**Comment by billy1624** [comment]
Hey @alper, it's just a placeholder when a entity doesn't have relation with other entities.

**Reply by alper** [reply]
Can we not have it in that case? I tried removing it but that errors.

**Comment by tyt2y3** [comment]
Umm the current type system requires a Relation associated type on Entity so we need these stub. We can either generate these inside a macro to hide it from plain sight, or rework the Entity type system.

**Comment by tyt2y3** [comment]
Basically whether we should have Entity and EntityWithRelation

**Comment by billy1624** [comment]
Actually, this works.

```rs
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}
```

will be expanded into

```rs
pub enum Relation {}

impl sea_orm::entity::RelationTrait for Relation {
    fn def(&self) -> sea_orm::entity::RelationDef {
        match self {
            _ => panic!("No RelationDef for Relation")
        }
    }
}
```

I will update the codegen and all related examples

**Reply by eduhenke** [reply]
you can also simply match with no arms `match *self {}` and it type-checks.
